src/main/java/com/google/starfish/servlets/UserRegistrationServlet.java

```diff
@@ -0,0 +1,128 @@
+
+  private static final Logger LOGGER = Logger.getLogger(UserRegistrationServlet.class.getName());
+  private static final String CLIENT_ID = System.getenv("CLIENT_ID");
+
+  @Override
+  public void doPost(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+    GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier.Builder(new NetHttpTransport(), JacksonFactory.getDefaultInstance())
+      .setAudience(Collections.singletonList(CLIENT_ID))
+      .build();
+
+    try {
+      GoogleIdToken idToken = verifier.verify(req.getParameter("idToken"));
+      if (idToken == null) {
+        res.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        res.getWriter().println("INVALID TOKEN ID");
+        return;
+      }
+
+      Payload payload = idToken.getPayload();
+
+      String userId = payload.getSubject();
+
+      // Get profile information from payload
+      String email = payload.getEmail();
+      String name = (String) payload.get("name");
+      String pictureUrl = (String) payload.get("picture");
+      String locale = (String) payload.get("locale");
+      String familyName = (String) payload.get("family_name");
+      String givenName = (String) payload.get("given_name");
+
+      DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");
+
+      try (Connection conn = pool.getConnection()) {
+        boolean userExists = checkIfUserExists(conn, userId);
+        // Do nothing if the user already exists
+        if (!userExists) {
+          String stmt =
+              "INSERT INTO users ( "
+                  + "id,"
+                  + "display_picture,"
+                  + "display_name,"
+                  + "date_joined,"
+                  + "email,"
+                  + "points,"
+                  + "school ) "
+            + "VALUES ( "
+                  + "?,"
+                  + "?,"
+                  + "?,"
+                  + "?,"
+                  + "?,"
+                  + "?,"
+                  + "? ); ";
+          try (PreparedStatement userStmt = conn.prepareStatement(stmt)) {
+            userStmt.setString(1, userId);
+            userStmt.setString(2, pictureUrl);
+            userStmt.setString(3, givenName);
+            userStmt.setDate(4, new Date(Calendar.getInstance().getTimeInMillis()));
+            userStmt.setString(5, email);
+            // User starts with 0 points
+            userStmt.setInt(6, 0);
+            userStmt.setNull(7, Types.VARCHAR);
+
+            // Finally, execute the statement. If it fails, an error will be thrown
+            userStmt.execute();
+
+          }
+        }
+
+        // Create new user session and save as cookie on client side
+        HttpSession newSession = req.getSession(true);
+        newSession.setAttribute("user_id", userId);
+        Cookie activeSession = new Cookie("sessionId", newSession.getId());
+        res.addCookie(activeSession);
+        res.sendRedirect("/profile.html");
```

Comment:
Would it be possible to return the user info to the web app (e.g. the number of points a user has or their display name)? Or do you think it'd be better to put that in a separate servlet?

Reply:
I think we'd put that in a separate servlet, so when you want the user info (maybe onLoad() on profile.html) it would then GET from the /users route and get the user who's id is stored in the active session on the cookie.

Reply:
I think during our discussion today we decided to have the servlet return a User object as a response, in addition to adding the cookie. Do you think you could just add a TODO in the code to say that? And then maybe it doesn't have to redirect to `profile.html` anymore?